infra/testApi.mjs

```diff
@@ -3,176 +3,129 @@ import { strict as assert } from 'node:assert';  // use strict equality everywhe
-assert.equal(improveResponse.status, 200)
-let redirect = improveResponse.url.split("/")
-const jobID = redirect[3].split(".")[0]
-// This test is a little flaky as the character count of the response is not consistent.
-// const improveHTML = await improveResponse.text()
-// const improveHTMLexpectedCount = 25871
-// assert.equal(improveHTML.length, improveHTMLexpectedCount, `HTML response character count should be ${improveHTMLexpectedCount} unless HTML changes.`)
-
-// timeline
-const timelineRSP = await fetch(makeEndpoint(`/timeline/${jobID}`), { method: 'GET' })
-assert.equal(timelineRSP.status, 201)
-const timeline = await timelineRSP.json()
-assert.equal(timeline.length > 0, true)
+const lots_of_vars = `(FPCore  (a b c d e f g h i j k l m n o p q r s t) 0)`
 
-// Test with a likely missing job-id
-const badTimelineRSP = await fetch(makeEndpoint("/timeline/42069"), { method: 'GET' })
-assert.equal(badTimelineRSP.status, 404)
-const check_missing_job = await fetch(makeEndpoint(`/check-status/42069`), { method: 'GET' })
-assert.equal(check_missing_job.status, 202)
+await simulateOdysseyCall(FPCoreFormula);
+await simulateOdysseyCall(FPCoreFormula2);
+await simulateOdysseyCall(FPCoreFormula3);
+await simulateOdysseyCall(lots_of_vars);
 
-// improve-start endpoint
-const URIencodedBody = "formula=" + encodeURIComponent(FPCoreFormula)
-const startResponse = await fetch(makeEndpoint("/api/start/improve"), {
-  method: 'POST',
-  headers: {
-    'Content-Type': 'application/x-www-form-urlencoded',
-  },
-  body: URIencodedBody
-})
-const testResult = (startResponse.status == 201 || startResponse.status == 202)
-assert.equal(testResult, true)
-const improveResultPath = startResponse.headers.get("location")
-let counter = 0
-let cap = 100
-// Check status endpoint
-let checkStatus = await fetch(makeEndpoint(improveResultPath), { method: 'GET' })
-/*
-This is testing if the /api/start/improve test at the beginning has been completed. The cap and counter is a sort of timeout for the test. Ends up being 10 seconds max.
-*/
-while (checkStatus.status != 201 && counter < cap) {
-  counter += 1
-  checkStatus = await fetch(makeEndpoint(improveResultPath), { method: 'GET' })
-  await new Promise(r => setTimeout(r, 100)); // ms
+// MARK: Analyze
+const expected_analyze = [[[14.97651307489794], "2.3"]]
+const analyze_result1 = await callAnalyze(FPCoreFormula, [[[
+  14.97651307489794
+], 0.12711304680349078]], false);
+
+const analyze_result2 = await callAnalyze(FPCoreFormula, [[[
+  14.97651307489794
+], 0.12711304680349078]], true);
+
+assert.deepEqual(analyze_result1.points, expected_analyze);
+assert.deepEqual(analyze_result2.points, expected_analyze);
+
+async function callAnalyze(fpcore, p_context, async_huh) {
+  const analyze_json = await getJSONFor('analyze', 'POST', JSON.stringify({
+    formula: fpcore, sample: p_context
+  }), async_huh);
+  assertIdAndPath(analyze_json);
```

Comment:
I think you call this every time you call `getJSONFor`, except for the mathjs translator.  Move into `getJSONFor`?